lib/modules/platform/scm.ts

```diff
@@ -0,0 +1,34 @@
+import * as _git from '../../util/git';
+import type { PlatformScm } from './types';
+
+export const platformScmImpls = new Map<string, Partial<PlatformScm>>();
+//here you can register additional custom implementations..
+
+export const defaultGitScm: PlatformScm = {
+  isBranchBehindBase: _git.isBranchBehindBase,
+  isBranchModified: _git.isBranchModified,
+  isBranchConflicted: _git.isBranchConflicted,
+  branchExists: (branchName: string) =>
+    Promise.resolve(_git.branchExists(branchName)),
+  getBranchCommit: (branchName: string) =>
+    Promise.resolve(_git.getBranchCommit(branchName)),
+  deleteBranch: _git.deleteBranch,
+};
```

Comment:
I would prefer there to be a `GitScm` class which implements `PlatformScm` interface, instead of this pattern. What do you think?

Side note: I understand that this isn't what you set out to do, and this is just necessary boilerplate to add support for Gerrit. This could be an intermediate step between only supporting git, and supporting multiple SCMs. I don't intend to block your PR or ask you to make this change now.